controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -500,9 +500,25 @@ func (w *Workload) UpdateKubeProxyImageInfo(ctx context.Context, kcp *controlpla
 	if err != nil {
 		return err
 	}
-	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil &&
-		kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository != "" {
-		newImageName, err = containerutil.ModifyImageRepository(newImageName, kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository)
+	var imageRepository string
+	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil {
+		imageRepository = kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository
+	}
+
+	// Beginning with kubernetes v1.25, kubernetes images including kube-proxy get published to registry.k8s.io instead of k8s.gcr.io.
+	// This ensures that the imageRepositorycontrolplane/kubeadm/internal/workload_cluster_coredns.go gets patched to registry.k8s.io when upgrading from v1.24 or lower,
+	// but only if there was no imageRespository explicitly set at the KubeadmControlPlanes ClusterConfiguration.
+	// This follows the behavior of `kubeadm upgrade`.
+	parsedVersion, err := semver.ParseTolerant(kcp.Spec.Version)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse kubernetes version %q", kcp.Spec.Version)
+	}
```

Comment:
What about passing version to UpdateKubeProxyImageInfo, so we are consistent with other Update methods like e.g. UpdateCoreDNS etc etc.